src/pytest_inmanta_lsm/orchestrator_container.py

```diff
@@ -19,13 +20,91 @@
 from types import TracebackType
 from typing import List, Optional, Tuple, Type
 
+import requests
 from inmanta.config import LenientConfigParser
+from packaging import version
 
 LOGGER = logging.getLogger(__name__)
 
 DOCKER_COMPOSE_COMMAND = None
 
 
+def get_image_version(image: str) -> version.Version:
+    """
+    Get the product version from the container image tag.
+    """
+    matched = re.fullmatch(
+        r".*\/service-orchestrator:(?P<tag>(?P<version>\d+(\.\d+)*)(-dev|-rc|-dev-ng)?|dev|dev-ng)",
+        image,
+    )
+    if matched is None:
+        raise ValueError(f"Unsupported image format: {image}")
+
+    tag = matched.group("tag")
+    v = matched.group("version")
+
+    if tag in ["dev", "dev-ng"]:
+        # The is the latest dev version of the product
+        return version.Version("8.dev")
```

Comment:
Any suggestion on how to keep this pointing to the next major of the product?

Reply:
Disclaimer: very hackish:

```
import requests


def get_latest_released_iso() -> int:
    """
    Calls "https://docs.inmanta.com/inmanta-service-orchestrator/latest/" url
    and relies on the fact that there's a redirection (?) to
    "https://docs.inmanta.com/inmanta-service-orchestrator/7/index.html"
    with the proper latest released iso version in there.
    """
    url = "https://docs.inmanta.com/inmanta-service-orchestrator/latest/"
    resp = requests.get(url)

    resp.raise_for_status()
    return int(resp.url.removeprefix("https://docs.inmanta.com/inmanta-service-orchestrator/").removesuffix("/index.html"))


latest_dev_iso = get_latest_released_iso() + 1

print(latest_dev_iso)
```

Reply:
I had something like that in mind but didn't dare doing it.  But if I am not the only one thinking about it it might be worth it :p